make/image.mk

```diff
@@ -0,0 +1,110 @@
+# This is a wrapper to build and push docker image
+#
+# All make targets related to docker image are defined in this file.
+
+# Set MAKE_IN_DOCKER=1 as an environment variable to run `make` inside
+# a Docker container with preinstalled tools.
+
+DOCKER_BUILDER_IMAGE ?= envoyproxy/gateway-dev-builder
+DOCKER_BUILDER_TAG ?= latest
+DOCKER_BUILD_CMD ?= DOCKER_BUILDKIT=1 docker build
+DOCKER_RUN_CMD ?= docker run \
+		  --rm \
+		  -t \
+		  -v /var/run/docker.sock:/var/run/docker.sock \
+		  -v ${PWD}:/workspace
+
+%:
+ifeq ($(MAKE_IN_DOCKER), 1)
+# Build builder image
+	@$(DOCKER_BUILD_CMD) -t $(DOCKER_BUILDER_IMAGE):$(DOCKER_BUILDER_TAG) - < ../tools/docker/Dockerfile
+# Run make inside the builder image
+# Run with MAKE_IN_DOCKER=0 to eliminate an infinite loop
+	@$(DOCKER_RUN_CMD) $(DOCKER_BUILDER_IMAGE):$(DOCKER_BUILDER_TAG) MAKE_IN_DOCKER=0 $@
+endif
+
+# Docker variables
+# REGISTRY is the image registry to use for build and push image targets.
+REGISTRY ?= docker.io/envoyproxy
+# IMAGE is the image URL for build and push image targets.
+IMAGE ?= ${REGISTRY}/gateway-dev
+# Tag is the tag to use for build and push image targets.
+TAG ?= $(REV)
+
+DOCKER := docker
+DOCKER_SUPPORTED_API_VERSION ?= 1.32
+
+ifdef HTTP_PROXY
+_DOCKER_BUILD_EXTRA_ARGS += --build-arg HTTP_PROXY=${HTTP_PROXY}
+endif
+
+ifneq ($(EXTRA_ARGS), )
+_DOCKER_BUILD_EXTRA_ARGS += $(EXTRA_ARGS)
+endif
+
+# Determine image files by looking into build/docker/*/Dockerfile
+IMAGES_DIR ?= $(wildcard ${ROOT_DIR}/docker/*)
+# Determine images names by stripping out the dir names
+IMAGES ?= envoy-gateway
+IMAGE_PLATFORMS ?= linux_amd64 linux_arm64 
+
+ifeq (${IMAGES},)
+  $(error Could not determine IMAGES, set ROOT_DIR or run in source dir)
+endif
+
+.PHONY: image.verify
+image.verify:
+	$(eval API_VERSION := $(shell $(DOCKER) version | grep -E 'API version: {1,6}[0-9]' | head -n1 | awk '{print $$3} END { if (NR==0) print 0}' ))
+	$(eval PASS := $(shell echo "$(API_VERSION) > $(DOCKER_SUPPORTED_API_VERSION)" | bc))
+	@if [ $(PASS) -ne 1 ]; then \
+		$(DOCKER) -v ;\
+		echo "Unsupported docker version. Docker API version should be greater than $(DOCKER_SUPPORTED_API_VERSION)"; \
+		exit 1; \
+	fi
+
+.PHONY: image.build
+image.build: image.verify  $(addprefix image.build., $(addprefix $(IMAGE_PLAT)., $(IMAGES)))
+
+.PHONY: image.build.multiarch
+image.build.multiarch: image.verify  $(foreach p,$(IMAGE_PLATFORMS),$(addprefix image.build., $(addprefix $(p)., $(IMAGES))))
+
+.PHONY: image.build.%
+image.build.%: go.build.%
+	$(eval IMAGE := $(COMMAND))
```

Comment:
I think `image.build` and `image.push` can be simplified using the current targets
https://github.com/envoyproxy/gateway/blob/f95b747d961f643bb8be2d1fac2cfa4ac32194cc/Makefile.targets.mk#L31
this eliminates the need to create and use `bin/tmp`

for multi arch you might need to use buildx similar to https://github.com/envoyproxy/envoy-build-tools/blob/main/build_container/docker_build_linux.sh or you can skip that target for now, and we can add it later.
the Github Action in the CI uses buildx internally